src/main/java/com/mlaf/hu/decisionagent/DecisionAgent.java

```diff
@@ -0,0 +1,102 @@
+package com.mlaf.hu.decisionagent;
+
+import com.mlaf.hu.decisionagent.behavior.ReceiveBehavior;
+import com.mlaf.hu.decisionagent.behavior.RegisterBehavior;
+import com.mlaf.hu.decisionagent.behavior.UpdateBehavior;
+import com.mlaf.hu.helpers.JadeServices;
+import com.mlaf.hu.models.*;
+import jade.core.AID;
+import jade.core.Agent;
+import jade.domain.DFService;
+import jade.util.Logger;
+
+import javax.xml.bind.JAXB;
+import java.io.StringReader;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public abstract class DecisionAgent extends Agent {
+    private static final String SERVICE_NAME = "DECISION-AGENT";
+    private static final int MAX_READINGS = 100;
+    public static java.util.logging.Logger decisionAgentLogger = Logger.getLogger("DecisionAgentLogger");
+    public HashMap<AID, InstructionSet> sensorAgents = new HashMap<>();
+
+    public DecisionAgent() {
+        super();
+    }
+
+    @Override
+    protected void setup() {
+        try {
+            JadeServices.registerAsService(SERVICE_NAME, "decision-agent", null, null, this);
+            addBehaviour(new RegisterBehavior(this));
+            addBehaviour(new ReceiveBehavior(this));
+            addBehaviour(new UpdateBehavior(this));
+        } catch (Exception e) {
+            DecisionAgent.decisionAgentLogger.log(Logger.SEVERE, "Could not initialize BrokerAgent", e);
+            System.exit(1);
+        }
+    }
+
+    protected void takeDown() {
+        try {
+            DFService.deregister(this);
+        } catch (Exception ignore) {
+        }
+    }
+
+    public void registerSensorAgent(AID sensoragent, InstructionSet instructionset) {
+        this.sensorAgents.put(sensoragent, instructionset);
+    }
+
+    public abstract void unregisterSensorAgent(AID sensoragent);
```

Comment:
Deze functie wordt nooit gebruikt?

Reply:
True, het is een abstract method die nog in de dummy of elke andere inheritance geïmplementeerd moet worden. Ik ga zo (20:29) beginnen aan de dummy decision agent.

Reply:
snap ik, maar voor zo ver ik zie wordt de methode ook nergens aangeroepen? Of moet de implementatie dit zelf doen?

Reply:
Dat laatste. Dit vraagt nog wel om een aanpassing, I agree. Want nu zou een sensoragent nooit verwijderd worden.

Reply:
Kijk dit is dus een mooie methode om aan te roepen als de instructie set niet te lezen is. Verder moet je zorgen dat de standaard jade dingen zelf afgehandeld worden (eventueel agent killen, mDNS unregistreren, uit AMS verwijderen enz). Alleen éxtra acties dus in het framework daar voor. (dus wel: éérst extra acties, daarna standaard unregister, anders is de agent natuurlijk al weg).